File: Exemplos/E06/pythonSqlite/teste_sqlite.py
import sqlite3

# Conectar ao banco de dados SQLite (ou criar se ele não existir)
conn = sqlite3.connect('exemplo.db')

# Criar um objeto cursor
cursor = conn.cursor()

# Criar uma nova tabela
cursor.execute('''CREATE TABLE IF NOT EXISTS usuarios
               (id INTEGER PRIMARY KEY, nome TEXT, idade INTEGER)''')

# Inserir dados na tabela
cursor.execute("INSERT INTO usuarios (nome, idade) VALUES ('Alice', 30)")
cursor.execute("INSERT INTO usuarios (nome, idade) VALUES ('Bob', 25)")

# Salvar (commit) as alterações
conn.commit()

# Consultar os dados
cursor.execute("SELECT * FROM usuarios")

# Buscar e imprimir os resultados
print(cursor.fetchall())

# Fechar a conexão
conn.close()

File: Exemplos/E06/goSqlite/teste_sqlite.go
package main

import (
	"database/sql"
	_ "github.com/mattn/go-sqlite3"
	"log"
)

func main() {
	db, _ := sql.Open("sqlite3", "./exemplo.db")
	defer db.Close() // Defer Closing the database

	// Criando a tabla
	sqlStmt := `
  CREATE TABLE IF NOT EXISTS usuarios
  (id INTEGER PRIMARY KEY, nome TEXT, idade INTEGER)
  `
	// Preparando o sql statement de forma segura
	command, err := db.Prepare(sqlStmt)
	if err != nil {
		log.Fatal(err.Error())
	}

	// Executando o comando sql
	command.Exec()

	// Criando uma função para inserir usuários
	insertUser := func(db *sql.DB, nome string, idade int) {
		stmt := `INSERT INTO usuarios(nome, idade) VALUES (?, ?)`
		statement, err := db.Prepare(stmt)
		if err != nil {
			log.Fatalln(err.Error())
		}
		_, err = statement.Exec(nome, idade)
		if err != nil {
			log.Fatalln(err.Error())
		}
	}

	insertUser(db, "Alice", 30)
	insertUser(db, "Bob", 25)

	displayUsers(db)
}

func displayUsers(db *sql.DB) {
	row, err := db.Query("SELECT * FROM usuarios ORDER BY nome")
	if err != nil {
		log.Fatal(err)
	}
	defer row.Close()
	for row.Next() {
		var id int
		var name string
		var age string
		row.Scan(&id, &name, &age)
		log.Println("Usuário: ", id, " ", name, " ", age)
	}
}
